src/main/scala/org/ergoplatform/http/api/ApplicationEntities.scala

```diff
@@ -0,0 +1,62 @@
+package org.ergoplatform.http.api
+
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder, HCursor, Json}
+import org.ergoplatform.ErgoBox.BoxId
+import org.ergoplatform.JsonCodecs
+import org.ergoplatform.settings.Algos
+import org.ergoplatform.wallet.Constants.ApplicationId
+import scorex.crypto.authds.ADKey
+
+/**
+  * Application-related classes and objects needed for API
+  */
+object ApplicationEntities {
+
+  case class ApplicationIdWrapper(appId: ApplicationId)
+
+  object ApplicationIdWrapper extends ApiCodecs {
+
+    implicit val applicationIdWrapperEncoder: Encoder[ApplicationIdWrapper] = { appStatus =>
```

Comment:
Why writing this boilerplate? Most of the codecs can be auto derived.

Reply:
Good catch! Made auto-derivation for ScanIdWrapper and ScanIdBoxId .